### DISCUSSION TITLE
Can I use this with Rust?

### DISCUSSION THREAD

**Discussion by JimLynchCodes** [typegoose/mongodb-memory-server#608] [Q&A]
Hi, I have previously used this in a nodejs project and liked how I could write unit tests that just got fake data from a fake db without me having to mock the mongo lib functions...

I am now working on a project using the [rust-mongo-driver](https://docs.mongodb.com/drivers/rust/) and am in need of something like this for those unit tests!!

**Comment by hasezoey** [comment]
From what i know from Rust, it does not (by default anyway) provide any functions like `beforeEach` / `beforeAll` for tests, so it would be kinda annoying to have a setup before every test

if you already use nodejs with your rust projects (like for semantic-release or husky or commitlint or something like that) and also use a CI system, then depending on the CI system you may have this library in a nodejs process running in the background, but if your CI does not support background jobs (like Github CI?) or just dont wanna have it running all the time, i would recommend to just use the download functionality of this library and manually start / stop the database, which can be directly called by using `MongoBinaryDownload` (after creating a instance of the class, call `getMongodPath`)

**Reply by JimLynchCodes** [reply]
thanks @hasezoey 

It is very strange to me that Rust decided not to have "beforeEach" although it seems like there are [hacky ways to do it](https://stackoverflow.com/questions/58006033/how-to-run-setup-code-before-any-tests-run-in-rust)...

Ideally I would be able to get this working on ci in a way that does not require a process running all the time.

When you say "using MongoBinaryDownload"... do you mean use it in a shell script before running the tests? Or actually import it in my Rust code with the "use" keyword? 🤔

Thanks!

**Reply by hasezoey** [reply]
> When you say "using MongoBinaryDownload"... do you mean use it in a shell script before running the tests? Or actually import it in my Rust code with the "use" keyword? thinking

i meant to make a small script that runs before your tests that calls the class and its functions to download it

example:
```js
#!/usr/bin/node

const mbd = require('mongodb-memory-server-core/lib/internal/MongoBinaryDownload');

(async () => {
  const mbd_instance = new mbd.MongoBinaryDownload({}); // options
  const binary_path = await mbd_instance.getMongodPath();
})();
```

note: the script above it from memory, not test or validated

**Reply by JimLynchCodes** [reply]
thanks frand, I will try this 👍
